src/backend/utils/adt/selfuncs.c

```diff
@@ -1281,6 +1281,8 @@ patternsel(PG_FUNCTION_ARGS, Pattern_Type ptype, bool negate)
 			Selectivity heursel;
 			Selectivity prefixsel;
 
+			Assert(prefix);
```

Comment:
I think that instead of asserting here, you should have removed the condition here https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/utils/adt/selfuncs.c#L1211 and there https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/utils/adt/selfuncs.c#L1342, and also added an assertion before the removed first condition.